WordPress/gutenberg
# Move Global Styles APIs to @wordpress/block-editor

## What?
- Moves `useSetting` in `edit-site` → `useGlobalSetting` in `block-editor`.
- Moves `useStyle` in `edit-site` → `useGlobalStyle` in `block-editor`.
- Moves `useGlobalStylesReset`, `useGlobalStylesOutput` and `GlobalStylesContext` in `edit-site` to `block-editor` as is.
- `lock()`s the above APIs within `wp.blockEditor.experiments`.
- The UI components and code that references the `global-styles` CPT remain in `@wordpress/edit-site`.

## Why?
More and more there is a need for blocks to access Global Styles. For example we want to:

- Show to the user what the default value will be (e.g. in a `placeholder`) is if a control is left empty ([context](https://github.com/WordPress/gutenberg/pull/46894))
- Display recently created style variations in the block inspector ([context](https://github.com/WordPress/gutenberg/pull/46343))
- Allow editing global styles from block inspector using _Push to Global Styles_ button ([context](https://github.com/WordPress/gutenberg/pull/46446))

https://github.com/WordPress/gutenberg/pull/46894 and https://github.com/WordPress/gutenberg/pull/34178 address this need by adding `useStyle` (to compliment `useSetting`) to `@wordpress/block-editor`. It's awkward though because any implementation of `useStyle` needs `getValueFromVariable` to make sense of CSS variables. This is an implementation detail of and not something that I think should be exposed from `@wordpress/block-editor`.

Instead we can move the high level Global Styles APIs to `@wordpress/block-editor` and make Global Styles a first class citizen of the block editor. This seems inevitable to me and to some extent has already begun. For example the block editor has a `__experimentalFeatures` setting which contains `mergedConfig.settings` and the block settings REST API has `__experimentalStyles` which contains `mergedConfig.styles`.

## Testing Instructions
There's no modified or new functionality here so just need to check for regressions. 

1. Go through each of the panels in Appearance → Editor → Styles and verify that there are no errors.
2. Test modifying and saving a few different global styles.
3. Test that 'Push globally' button still works by modifying the styles of a block and pressing 'Push globally' in the 'Advanced' panel.
